### DISCUSSION TITLE
[BUG] `too large device id 11, use 10 instead`

### DISCUSSION THREAD

**Discussion by zyf8827** [dashingsoft/pyarmor#1402] [General]
As the [document](https://pyarmor.readthedocs.io/en/stable/how-to/register.html#using-group-license) says, group license supports 100 offline devices. But when I tried to register more than 10 devices, I got the following error:

```
# pyarmor reg -g 11 pyarmor-regfile-6051.zip
INFO     Python 3.9.17
INFO     Pyarmor 8.2.9 (group), 006051, zeta
INFO     Platform linux.x86_64
INFO     register device file ".pyarmor/group/pyarmor-group-device.11"
INFO     use group license "pyarmor-regfile-6051.zip"
INFO     send request to server
ERROR    too large device id 11, use 10 instead
```

This is my env info:
```
$ pyarmor gen
INFO     Python 3.9.17
INFO     Pyarmor 8.2.9 (group), 006051, zeta
INFO     Platform linux.x86_64

```

**Comment by jondy** [comment]
Why not `pyarmor reg -g 10 ...` as the error message has told you the reason?

**Reply by zyf8827** [reply]
Group 1 to 10 have been used in 10 different devices. I tried to reuse group and got the following error:

![image](https://github.com/dashingsoft/pyarmor/assets/5118661/890bc185-fac4-4b00-9788-7a9676c1351d)

**Reply by zyf8827** [reply]
It's odd, I tried to register device 11 and got a successful message today.

![image](https://github.com/dashingsoft/pyarmor/assets/5118661/7626e025-5aa4-4c1f-91a9-9fe6f8257d2c)
